### PR TITLE
Forward SIGTERM and SIGINT to wrapped process from entrypoint

### PR DESCRIPTION
**Pull request by stevekuznetsov**
When the kubelet begins graceful termination of the entrypoint
container, we need to forward the signal we get to the wrapped process
and begin the graceful termination of the wrapped process as well so the
process has time to do cleanup before getting killed.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

/area prow
/kind bug
fixes https://github.com/kubernetes/test-infra/issues/7738
depends on #7654 

/cc @BenTheElder @kargakis 
/assign @cjwagner @fejta

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
```
W0418 15:16:04.034] Run: ('bazel', 'query', '--keep_going', '--noshow_progress', "kind(test, rdeps(//..., //...)) except attr('tags', 'manual', //...)")
W0418 15:16:04.043] INFO: $TEST_TMPDIR defined: output root default is '/bazel-scratch/.cache/bazel' and max_idle_secs default is '15'.
W0418 15:26:15.106] ERROR: /workspace/k8s.io/test-infra/prow/cmd/gerrit/BUILD.bazel:7:1: no such package '@python//image': Pull command failed: Timed out (/usr/bin/python /bazel-scratch/.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/puller/file/puller.par --directory /bazel-scratch/.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/python/image --name index.docker.io/library/python@sha256:8bfeec8f8ba3aaeea918a0198f4b1c7c9b2b39e26f399a7173229dfcef76fc1f) and referenced by '//prow/cmd/gerrit:image'

W0418 15:26:15.116] WARNING: errors encountered while loading target '//prow/cmd/gerrit:image'
W0418 15:26:15.116] WARNING: errors encountered while loading target '//prow/cmd/gerrit:push'
W0418 15:26:15.323] ERROR: /workspace/k8s.io/test-infra/prow/cmd/gerrit/BUILD.bazel:7:1: no such package '@python//image': Pull command failed: Timed out (/usr/bin/python /bazel-scratch/.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/puller/file/puller.par --directory /bazel-scratch/.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/python/image --name index.docker.io/library/python@sha256:8bfeec8f8ba3aaeea918a0198f4b1c7c9b2b39e26f399a7173229dfcef76fc1f) and referenced by '//prow/cmd/gerrit:image'

W0418 15:26:15.380] ERROR: Evaluation of query "rdeps(//..., //...)" failed: errors were encountered while computing transitive closure
```

@BenTheElder @ixdy  :(

**Comment by shyamjvs** [comment]
Sorry that label was by accident.

**Comment by BenTheElder** [comment]
Caching issues shouldn't cause no log uploads, that's something else.

**Comment by fejta** [comment]
Gubernator caching issue? The files are there: http://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/7759/pull-test-infra-lint/1676/ /cc @rmmh

**Comment by ixdy** [comment]
@fejta different job ID, 1674 vs 1676. I'm guessing the first got killed/rescheduled for some reason.

File: prow/entrypoint/run.go [hunk 1]
```diff
@@ -76,7 +77,13 @@ func (o Options) Run() error {
 		return fmt.Errorf("could not start the process: %v", err)
 	}
 
+	// if we get asked to terminate we need to forward
+	// that to the wrapped process as if it timed out
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt, syscall.SIGTERM)
+
 	timeout := time.Duration(optionOrDefault(o.Timeout, DefaultTimeout))
+	gracePeriod := time.Duration(optionOrDefault(o.GracePeriod, DefaultGracePeriod))
```
Comment (cjwagner) [inline prow/entrypoint/run.go]: The return value is already a `time.Duration`.

**Comment by stevekuznetsov** [comment]
Rebased to address nit and fix merge conflict.

/hold cancel

**Pull request merged**